### PR TITLE
chore: rename 'progressive rollouts' to 'progressive syncs'

### PR DESCRIPTION
**Pull request by crenshaw-dev**
Thanks to @christianh814 for the idea. I think this is clearer and avoids clash with Argo Rollouts. The rename is also endorsed by @wmgroot (who wrote the feature) and @zachaller (who is an Approver on Rollouts).

### CODE REVIEW

**Review by mbevc1 — APPROVED** [review]

**Review by zachaller — APPROVED** [review]

**Review by christianh814 — APPROVED** [review]
LGTM!

**Comment by wmgroot** [comment]
LGTM.

**Review by wanghong230 — APPROVED** [review]

File: docs/operator-manual/applicationset/Progressive-Syncs.md [hunk 1]
```diff
@@ -1,21 +1,21 @@
...
-As an experimental feature, progressive rollouts must be explicitly enabled, in one of these ways.
+## Enabling Progressive Syncs
+As an experimental feature, progressive syncs must be explicitly enabled, in one of these ways.
 
-1. Pass `--enable-progressive-rollouts` to the ApplicationSet controller args.
-1. Set `ARGOCD_APPLICATIONSET_CONTROLLER_ENABLE_PROGRESSIVE_ROLLOUTS=true` in the ApplicationSet controller environment variables.
-1. Set `applicationsetcontroller.enable.progressive.rollouts: true` in the ArgoCD ConfigMap.
+1. Pass `--enable-progressive-syncs` to the ApplicationSet controller args.
+1. Set `ARGOCD_APPLICATIONSET_CONTROLLER_ENABLE_PROGRESSIVE_SYNCS=true` in the ApplicationSet controller environment variables.
+1. Set `applicationsetcontroller.enable.progressive.syncs: true` in the Argo CD ConfigMap.
```
Comment (drpaneas) [inline docs/operator-manual/applicationset/Progressive-Syncs.md:18]:
```suggestion
1. Pass `--enable-progressive-syncs` to the ApplicationSet controller args.
2. Set `ARGOCD_APPLICATIONSET_CONTROLLER_ENABLE_PROGRESSIVE_SYNCS=true` in the ApplicationSet controller environment variables.
3. Set `applicationsetcontroller.enable.progressive.syncs: true` in the Argo CD ConfigMap.
```

File: cmd/argocd-applicationset-controller/commands/applicationset_controller.go [hunk 2]
```diff
@@ -207,7 +207,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().StringVar(&cmdutil.LogFormat, "logformat", env.StringFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_LOGFORMAT", "text"), "Set the logging format. One of: text|json")
 	command.Flags().StringVar(&cmdutil.LogLevel, "loglevel", env.StringFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_LOGLEVEL", "info"), "Set the logging level. One of: debug|info|warn|error")
 	command.Flags().BoolVar(&dryRun, "dry-run", env.ParseBoolFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_DRY_RUN", false), "Enable dry run mode")
-	command.Flags().BoolVar(&enableProgressiveRollouts, "enable-progressive-rollouts", env.ParseBoolFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_ENABLE_PROGRESSIVE_ROLLOUTS", false), "Enable use of the experimental progressive rollouts feature.")
+	command.Flags().BoolVar(&enableProgressiveSyncs, "enable-progressive-rollouts", env.ParseBoolFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_ENABLE_PROGRESSIVE_SYNCS", false), "Enable use of the experimental progressive syncs feature.")
```
Comment (drpaneas) [inline cmd/argocd-applicationset-controller/commands/applicationset_controller.go:210]:
```suggestion
	command.Flags().BoolVar(&enableProgressiveSyncs, "enable-progressive-syncs", env.ParseBoolFromEnv("ARGOCD_APPLICATIONSET_CONTROLLER_ENABLE_PROGRESSIVE_SYNCS", false), "Enable use of the experimental progressive syncs feature.")
```

**Review by drpaneas — COMMENTED** [review]
I think you need to change this as well.

**Review by leoluz — APPROVED** [review]
LGTM

**Comment by crenshaw-dev** [comment]
Cherry-picked onto release-2.6 for 2.6.0.

**Pull request merged**